src/halfpipe/tui/specialized_widgets/non_bids_file_itemization.py

```diff
@@ -0,0 +1,359 @@
+        self.delete_button = delete_button
+        self.pattern_class = None if pattern_class is None else pattern_class
+        self.title = "Not implemented yet"
+        if self.pattern_class is not None:
+            self.title = self.pattern_class.header_str
+            if self.pattern_class.next_step_type is not None:
+                self.pattern_class.callback = self.callback_func
+            self.pattern_class.id_key = id
+
+        self.load_object = load_object
+        self.border_title = "id: " + str(id)
+        self.from_edit = False
+        if message_dict is not None and callback_message is None:
+            self.callback_message = self.prettify_message_dict(message_dict)
+        else:
+            self.callback_message = callback_message
+
+        self.pattern_match_results = {"file_pattern": "", "message": "Found 0 files.", "files": []}
+        self.execute_pattern_class_on_mount = execute_pattern_class_on_mount
+
+    def prettify_message_dict(self, message_dict):
+        info_string = Text("")
+        for key in message_dict:
+            # if there is only one item, we do not separate items on new lines
+            if len(message_dict[key]) <= 1:
+                sep_char = ""
+                separ_line = "-" * (len(key) + len(message_dict[key]) + 3)
+            else:
+                sep_char = "\n"
+                separ_line = "-" * (max([len(s) for s in [key] + message_dict[key]]) + 3)
+            message_value = ""
+            for message in message_dict[key]:
+                message_value += message + " " if message.endswith("\n") else message + "\n"
+            info_string += Text(humanize(key) + ": " + sep_char, style="bold green") + Text(
+                message_value + separ_line, style="white"
+            )
+        return info_string
+
+    def callback_func(self, message_dict):
+        self.callback_message = self.prettify_message_dict(message_dict)
+
+    def compose(self):
+        yield HorizontalScroll(Static("Edit to enter the file pattern", id="static_file_pattern"))
+        with Horizontal(id="icon_buttons_container"):
+            yield Button(" ℹ", id="info_button", classes="icon_buttons")
+
+            yield Button("🖌", id="edit_button", classes="icon_buttons")
+            yield Button("👁", id="show_button", classes="icon_buttons")
+            if self.delete_button:
+                yield Button("❌", id="delete_button", classes="icon_buttons")
+
+    def on_mount(self) -> None:
+        if self.load_object is None:
+            self.get_widget_by_id("edit_button").tooltip = "Edit"
+            if self.delete_button:
+                self.get_widget_by_id("delete_button").tooltip = "Delete"
+            if self.pattern_class is not None:
+                self.app.push_screen(
+                    PathPatternBuilder(
+                        path="/home/tomas/github/ds005115/sub-01/ses-01/func/sub-01_ses-01_task-rest_bold.nii.gz",
```

Comment:
I'm not sure if this is a good initial value

Reply:
now it is just the root '/'